index.js

```diff
@@ -40,3 +40,36 @@ formElement.addEventListener('submit', (event) => {
     smallElement.innerText = 'please type all email charaters in lowercase. form was not submitted';
   }
 });
+
+const store = (key, value) => {
+  localStorage.setItem(key, value);
+};
+
+const saveForm = (event) => {
+  const formdata = {};
+  event.preventDefault();
+  if (formElement.elements.name.value) {
+    formdata.name = formElement.elements.name.value;
+  }
+  if (formElement.elements.email.value) {
+    formdata.email = formElement.elements.email.value;
+  }
+  if (formElement.elements.text.value) {
+    formdata.text = formElement.elements.text.value;
+  }
+  store('formdata', JSON.stringify(formdata));
+};
+
+formElement.elements.name.addEventListener('input', saveForm);
+formElement.elements.email.addEventListener('input', saveForm);
+formElement.elements.text.addEventListener('input', saveForm);
+
+const showSavedData = () => {
+  if (localStorage.getItem('formdata') !== true) return;
+  const userData = JSON.parse(localStorage.getItem('formdata'));
+  formElement.elements.name.value = userData.name;
+  formElement.elements.email.value = userData.email;
+  formElement.elements.text.value = userData.text;
```

Comment:
- [ ] Well done with setting up your local storage to store user details for more scintillating user experience. The only issue we have now is that the data do not persist, and are not being retrieved after the page reloads or refreshes. as the require posits that it functions.
- This might be due to the fact that `showSavedData` function has no `event` to call or trigger it. Try adding an `eventListener` of `onload` i.ee whenever page reloads, call this function and prefill with available form data in the storage.
- Just to show you that data is saved, but the updating of form fields is not happening:
![image](https://user-images.githubusercontent.com/99715640/224346238-6a02e8f5-c4a9-4de0-b7b9-9a657851eeae.png)